nav2_behavior_tree/include/nav2_behavior_tree/behavior_tree_engine.hpp

```diff
@@ -69,6 +70,18 @@ class BehaviorTreeEngine
   }
 
 protected:
+  // Methods used to register as (simple action) BT nodes
+  BT::NodeStatus globalLocalizationServiceRequest();
```

Comment:
`globalLocalizationServiceRequest` and `initialPoseReceived` arent used, that was part of my proposal to just delete them. They break important Navigation encapsulation concepts.

Reply:
Ok, if they aren't used, I can delete them.

Reply:
The navigator shouldn't be tightly coupled with the positioning system, hooks are OK but the core navigator code *in my opinion* shouldn't contain positioning specific items. Since this is the "core" of the BT (and in future loading of nodes) this seems like exactly the place this shouldn't be. Probably as a utility? 

Yes the factory items are all good since we haven't gotten to plugin-izing things.

Reply:
How about the localization-related nodes as a separate plugin from the navigation-related nodes? For now, both would be registered in the BehaviorTreeEngine, but soon, they would be separated out so that the BehaviorTreeEngine is generic and each plug-in loads its own node types.

Reply:
I don't think localization-based nodes should exist except for checking-state to trigger something like "stop navigator, I'm delocalized an entering a localization-driven motion mode to re-establish localization". The navigator's job is to take positioning and go to a place according to some rule set that the BT defines. I think that breaks the encapsulation of a pure-navigator. 

It seems to me it would make most sense if the autolocalization mode was an action server that could be called from a node that checks localization quality and triggers, but it itself shouldn't really be in the Navigators behavior tree

Reply:
Ok, I understand your point about trying to keep the navigation BT "pure." However, if we made the autolocalization an action server and chose to implement this action server with a BT (similar to how the navigation is implemented using a BT), we'd still need to define the localization-related BT nodes (presumably using a plugin-model) for the auto-localization action server. We've then also introduced yet another node. How about instead using the XML include mechanism supported in the 3.0 library to move the auto-localization sub-tree into its own XML file. The net effect is that the navigation BT references only a single, external node/subtree. We have the benefit of encapsulation without the overhead of introducing another node and action server.

Reply:
Why would autolocalization need to be a behavior tree? The package is called `nav2_bt_navigator` and that has a set of expectations on it - trying to have a package that does everything is probably not a good way to go. 

Auto-localization is using experimental tech that only works for 1 robot model in simulation (safely), I dont think that is mature enough to justify making compromises for in design. If there's some hacky-way we get auto-localization into the stack, I think that's probably more appropriate since its not something most users will be using at this point as it won't work on most people's configurations.

Reply:
Reading back over this: the 2 fns in question both just call services or topics. Why do they have to be in the navigator? The autolocalization module should be more than capable of creating it own service clients, correct @mjeronimo